src/docfx/build/toc/TableOfContentsLoader.cs

```diff
@@ -37,34 +41,83 @@ internal class TableOfContentsLoader
             TableOfContentsParser parser,
             MonikerProvider monikerProvider,
             DependencyMapBuilder dependencyMapBuilder,
-            ContentValidator contentValidator)
+            ContentValidator contentValidator,
+            Config config,
+            ErrorBuilder errors,
+            DocumentProvider documentProvider)
         {
             _linkResolver = linkResolver;
             _xrefResolver = xrefResolver;
             _parser = parser;
             _monikerProvider = monikerProvider;
             _dependencyMapBuilder = dependencyMapBuilder;
             _contentValidator = contentValidator;
+            _errors = errors;
+            _documentProvider = documentProvider;
+            _joinTOCConfigs = config.JoinTOC is null
+                ? new Dictionary<string, JoinTOCConfig>()
+                : config.JoinTOC.Where(x => x.ReferenceToc != null).ToDictionary(x => PathUtility.Normalize(x.ReferenceToc!));
         }
 
         public (TableOfContentsNode node, List<Document> referencedFiles, List<Document> referencedTocs)
-            Load(ErrorBuilder errors, Document file)
+            Load(Document file)
         {
             return _cache.GetOrAdd(file.FilePath, _ =>
             {
                 var referencedFiles = new List<Document>();
                 var referencedTocs = new List<Document>();
-                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, errors);
+                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, _errors);
 
+                if (_joinTOCConfigs.TryGetValue(file.FilePath.Path, out var joinTOCConfig) && joinTOCConfig != null)
+                {
+                    if (joinTOCConfig.TopLevelToc != null)
+                    {
+                        node = JoinToc(node, joinTOCConfig.TopLevelToc);
+                    }
+                }
                 return (node, referencedFiles, referencedTocs);
             });
         }
 
+        private TableOfContentsNode JoinToc(TableOfContentsNode referenceToc, string topLevelTocFilePath)
```

Comment:
>referenceToc [](start = 64, length = 12)

topLevelToc might be a better name

Reply:
Would like to specify this is for transformation of `referenceToc` with skeleton `topLevelToc` corresponding to ops config